File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/messages/MessageAction.java
package io.github.tanguygab.conditionalactions.actions.types.messages;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

public class MessageAction extends Action {

    public MessageAction() {
        super("^(?i)(message|msg|tell):( )?");
    }

    @Override
    public String getSuggestion() {
        return "message: <message>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        match = parsePlaceholders(player, match, true);
        if (player instanceof Player p) p.sendMessage(match);
        else getPlugin().getServer().getConsoleSender().sendMessage(match);
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/messages/MessageAction.kt
package io.github.tanguygab.conditionalactions.actions.types.messages

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player

class MessageAction : Action("^(?i)(message|msg|tell):( )?".toRegex()) {
    override fun getSuggestion() = "message: <message>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val match = mm.deserialize(parsePlaceholders(player, match))
        if (player is Player) player.sendMessage(match)
        else plugin.server.consoleSender.sendMessage(match)
    }
}
